source/extensions/filters/http/ext_proc/config.cc

```diff
@@ -9,62 +9,41 @@ namespace Extensions {
 namespace HttpFilters {
 namespace ExternalProcessing {
 
-Http::FilterFactoryCb ExternalProcessingFilterConfig::createFilterFactoryFromProtoTyped(
+absl::StatusOr<Http::FilterFactoryCb>
+ExternalProcessingFilterFactory::createFilterFactoryFromProtoTyped(
     const envoy::extensions::filters::http::ext_proc::v3::ExternalProcessor& proto_config,
-    const std::string& stats_prefix, Server::Configuration::FactoryContext& context) {
+    const std::string& stats_prefix, DualInfo dual_info,
+    Server::Configuration::ServerFactoryContext& context) {
   const uint32_t message_timeout_ms =
       PROTOBUF_GET_MS_OR_DEFAULT(proto_config, message_timeout, DefaultMessageTimeoutMs);
   const uint32_t max_message_timeout_ms =
       PROTOBUF_GET_MS_OR_DEFAULT(proto_config, max_message_timeout, DefaultMaxMessageTimeoutMs);
   const auto filter_config = std::make_shared<FilterConfig>(
       proto_config, std::chrono::milliseconds(message_timeout_ms), max_message_timeout_ms,
-      context.scope(), stats_prefix,
-      Envoy::Extensions::Filters::Common::Expr::getBuilder(context.serverFactoryContext()),
-      context.serverFactoryContext().localInfo());
+      context.scope(), stats_prefix, dual_info.is_upstream_filter,
+      Envoy::Extensions::Filters::Common::Expr::getBuilder(context), context.localInfo());
 
-  return [filter_config, grpc_service = proto_config.grpc_service(),
-          &context](Http::FilterChainFactoryCallbacks& callbacks) {
+  return [filter_config, grpc_service = proto_config.grpc_service(), &context,
+          dual_info](Http::FilterChainFactoryCallbacks& callbacks) {
     auto client = std::make_unique<ExternalProcessorClientImpl>(
-        context.serverFactoryContext().clusterManager().grpcAsyncClientManager(), context.scope());
+        context.clusterManager().grpcAsyncClientManager(), dual_info.scope);
 
     callbacks.addStreamFilter(Http::StreamFilterSharedPtr{
         std::make_shared<Filter>(filter_config, std::move(client), grpc_service)});
   };
 }
 
 Router::RouteSpecificFilterConfigConstSharedPtr
-ExternalProcessingFilterConfig::createRouteSpecificFilterConfigTyped(
+ExternalProcessingFilterFactory::createRouteSpecificFilterConfigTyped(
     const envoy::extensions::filters::http::ext_proc::v3::ExtProcPerRoute& proto_config,
     Server::Configuration::ServerFactoryContext&, ProtobufMessage::ValidationVisitor&) {
   return std::make_shared<FilterConfigPerRoute>(proto_config);
 }
 
-Http::FilterFactoryCb
```

Comment:
Looks like the PR is still in draft mode. I just want to leave an early comment about removing this method.

Currently, it is used in match_delegate filter [like here](https://github.com/envoyproxy/envoy/blob/7c410102b1a97b439c8b86bc10c0c5059a408ca3/source/extensions/filters/http/composite/action.cc#L75) so that ext_proc filter can be created in match_delegate's per route case.

Thus, deleting this method will break that. We will need to come up with non-breaking solution along with recent refactor of factory/server context.